generators/client/templates/react/src/main/webapp/app/shared/reducers/locale.ts.ejs

```diff
@@ -35,7 +35,6 @@ export default (state: LocaleState = initialState, action): LocaleState => {
     case ACTION_TYPES.SET_LOCALE:
       const currentLocale = action.locale;
       if (state.currentLocale !== currentLocale) {
-        Storage.session.set('locale', currentLocale);
```

Comment:
I think its better to do this here instead of in header so that this is responsibility of this reducer and I think it doesnt change any behavior

Reply:
first time open the page, here will set the `locale` to `native lang`. when it reach the `get account info`, we are not able to use user preferred language anymore since the `locale` is not null. 

In this PR. the React and Angular share the same logic: only persist the `locale` when user select the language.